test/ScheduledChannelTest.cpp

```diff
@@ -0,0 +1,112 @@
+#include "pipert/ScheduledChannel.h"
+#include "pipert/Scheduler.h"
+#include "gtest/gtest.h"
+
+namespace{
+    class OtherPrinter {
+        public:
+        void Print(pipert::PacketToProcess<int> packet) {
+            std::cout << "Other: " << packet.data() << std::endl;
+        }
+        void Add(pipert::PacketToProcess<int> packet) {
+            std::cout << packet.data() << std::endl;
+            packet.data() = 11;
+        }
+    };
+
+} //namespace
+
+class ScheduledChannelTest : public ::testing::Test {
+    protected:
+    ::pipert::Scheduler scheduler_;
+    ::pipert::Timer::Time default_time_;
+    ::pipert::Timer::Time old_time_;
+    ::pipert::Timer::Time new_time_;
+    OtherPrinter op_;
+
+    ScheduledChannelTest() 
+        : scheduler_(1),
+        default_time_(1603716824924000),
+        old_time_(1603716857718209),
+        new_time_(1603716871368358) {
+    }
+
+    virtual ~ScheduledChannelTest() {
+    }
+
+    virtual void SetUp() {
+    }
+
+    virtual void TearDown() {
+    }
+
+};
+
+
+TEST_F(ScheduledChannelTest, ScheduledChannelCreationWithSchedulerBasicChannelInfoTest) {
+    // When
+    ::pipert::ScheduledChannel<int> channel = scheduler_.CreateScheduledChannel<int>("TestChannel", 2, nullptr, std::bind(&OtherPrinter::Print, &op_, std::placeholders::_1));
+    // Then
+    EXPECT_EQ(channel.GetName(), "TestChannel");
```

Comment:
This is testing pointer equality instead of string comparison. So with this, you are relying on the compiler doing a string constant merging where it realizes that TestChannel is only compiled once into the binary and the 2 pointers will be the same. So this may or may not happen. Try to use EXPECT_STR... something instead!

Reply:
Just to clarify: I know that in the implementation, we just put the const pointer away, so it will really be good on every platform. But this way, you are testing more than just the expected behavoir. You are testing implicitly something in the implementation.